libPhoneNumber/NBMetadataHelper.m

```diff
@@ -195,20 +203,25 @@ - (NBPhoneMetaData *)shortNumberMetadataForRegion:(NSString *)regionCode
 
     regionCode = [regionCode uppercaseString];
 
-    if ([_cachedShortNumberMetaDataKey isEqualToString:regionCode]) {
-        return _cachedShortNumberMetadata;
+  @synchronized(_shortNumberMetadataCache) {
```

Comment:
NSCache is "thread" safe in that you can set/get without wrapping in a lock. See the NSCache documentation. So please remove the @sync (and maybe improve your performance ;-) )

Reply:
Done